google-calendar-like-interface/index.html

```diff
@@ -0,0 +1,261 @@
+        <div class="hour">15</div>
+        <div class="hour">16</div>
+        <div class="hour">17</div>
+        <div class="hour">18</div>
+        <div class="hour">19</div>
+        <div class="hour">20</div>
+      </div>
+      <div id="hours-7">
+        <div class="hour">1</div>
+        <div class="hour">2</div>
+        <div class="hour">3</div>
+        <div class="hour">4</div>
+        <div class="hour">5</div>
+        <div class="hour">7</div>
+        <div class="hour">8</div>
+        <div class="hour">9</div>
+        <div class="hour">10</div>
+        <div class="hour">11</div>
+        <div class="hour">12</div>
+        <div class="hour">13</div>
+        <div class="hour">14</div>
+        <div class="hour">15</div>
+        <div class="hour">16</div>
+        <div class="hour">17</div>
+        <div class="hour">18</div>
+        <div class="hour">19</div>
+        <div class="hour">20</div>
+      </div>
+    </div>
+
+    <script>
+      var dragging;
+      var initialYCoordinate;
+      var initialBox;
+      var initialColumn;
+
+      document.querySelectorAll('.hour').forEach(function addMouseDownEvent(box) {
+        box.addEventListener('mousedown', function handleInitialClick(event) {
+          event.preventDefault();
+
+          dragging = true;
+          initialYCoordinate = event.pageY; // y coordinate where the click happened
+          initialBox = box;
+          initialColumn = box.parentElement;
+          document.querySelector('body').style.cursor = 'move';
+        });
+      });
+
+      document.addEventListener('mouseup', function stopDragging(event) {
+        event.preventDefault();
+        dragging = false;
+        document.querySelector('body').style.cursor = '';
+      });
+
+      function isBoxBelowInitialY(box) {
+        var top = box.offsetTop;
+        var bottom = top + box.offsetHeight;
+
+        return top > initialYCoordinate;
+      }
+
+      function isInitialBox(box) {
+        var top = box.offsetTop;
+        var bottom = top + box.offsetHeight;
+
+        return initialBox.offsetTop == top && initialBox.offsetTop + box.offsetHeight === bottom;
+      }
+
+      function hasBoxBeenIntersected(box, currentY) {
+        var top = box.offsetTop;
+
+        return currentY >= top;
+      }
+
+      function isAboveInitialY(box) {
+        var top = box.offsetTop;
+        var bottom = top + box.offsetHeight;
+
+        return bottom < initialYCoordinate;
+      }
+
+      function hasBeenIntersectedUp(box, currentY) {
+        var top = box.offsetTop;
+        var bottom = top + box.offsetHeight;
+
+        return currentY < bottom;
+      }
+
+      var direction;
+
+      document.addEventListener('mousemove', function handleMouseMove(event) {
+        var currentY = event.pageY;
+        direction = currentY >= initialYCoordinate ? 'down' : 'up';
+
+        if (dragging) {
+          document
+            .querySelectorAll(`#${initialColumn.id} .hour`)
```

Comment:
extract this into a variable, it's easier to read

```suggestion
            .querySelectorAll(boxesWithColumnId)
```